### PR TITLE
Tcprtt: add --byladdr/--byraddr

### PR DESCRIPTION
**Pull request by pacepi**
@brendangregg would you please review this PR?

### CODE REVIEW

**Comment by yonghong-song** [comment]
[buildbot, test this please]

File: tools/tcprtt_example.txt [hunk 1]
```diff
@@ -40,16 +40,65 @@ also shows unstable TCP RTT. So in this situation, we need to make sure the
...
+and remote address 192.168.122.100 and remote port 80.
 # ./tcprtt -i 1 -d 10 -m -a 192.168.122.200 -A 192.168.122.100 -P 80
 
 
+Tracing at server side, show each clients with its own histogram.
+For example, run tcprtt on a storage node to show initiators' rtt histogram:
+# ./tcprtt -i 1 -m --lport 3260 --byraddr
+Tracing TCP RTT... Hit Ctrl-C to end.
+
+srtt = Remote Address: 10.131.90.18
```
Comment (yonghong-song) [inline tools/tcprtt_example.txt:53]: Could you change the head to 
  Remote Address = 10.131.90.18
This is to conform to other bcc tool histogram output (e.g., biolatency.py).

**Review by yonghong-song — COMMENTED** [review]
Looks good. Thanks. Only one minor comments about output format.

**Comment by yonghong-song** [comment]
[buildbot, ok to test]

**Comment by brendangregg** [comment]
Looks good thanks, but I have a question: you've just renamed sport to lport (etc.). And these come from:

```
/** struct inet_sock - representation of INET sockets
 *
 * @sk - ancestor class
 * @pinet6 - pointer to IPv6 control block
 * @inet_daddr - Foreign IPv4 addr
 * @inet_rcv_saddr - Bound local IPv4 addr
 * @inet_dport - Destination port
 * @inet_num - Local port
 * @inet_saddr - Sending source
 * @uc_ttl - Unicast TTL
 * @inet_sport - Source port
 * @inet_id - ID counter for DF pkts
```

These are getting the right ports for both active and passive connections, right? Just checking. Do you also know what inet_num is? Since it says "Local port"! :-)

File: tools/tcprtt_example.txt (hunk 1 above)
Comment (pacepi) [reply tools/tcprtt_example.txt]: Fixed as you suggested.

**Comment by pacepi** [comment]
> Looks good thanks, but I have a question: you've just renamed sport to lport (etc.). And these come from:
> 
> ```
> /** struct inet_sock - representation of INET sockets
>  *
>  * @sk - ancestor class
>  * @pinet6 - pointer to IPv6 control block
>  * @inet_daddr - Foreign IPv4 addr
>  * @inet_rcv_saddr - Bound local IPv4 addr
>  * @inet_dport - Destination port
>  * @inet_num - Local port
>  * @inet_saddr - Sending source
>  * @uc_ttl - Unicast TTL
>  * @inet_sport - Source port
>  * @inet_id - ID counter for DF pkts
> ```
> 
> These are getting the right ports for both active and passive connections, right? Just checking. Do you also know what inet_num is? Since it says "Local port"! :-)

Because this tool sets probe at function 'tcp_rcv_established', so it always works for 'ESTABLISHED' connections. Source port is  'local port' for an established connection, isn't it? I'm not familiar enough, could you give me more hints?

**Comment by yonghong-song** [comment]
Searching the kernel source code, `inet_num` is defined as
```
#define inet_num sk.__sk_common.skc_num
```
`inet_num` is used in different ways, for tcp/udp sockets, it may represent and eventually assign to a local/src port, it may also assign to `tw_num` of a timewait socket, or protocol for raw socket, etc.

Yes, `tcp_rcv_established` does track established connection for both active and passive connections.
`netstat -a` uses local address and foreign address
```
Active Internet connections (servers and established)
Proto Recv-Q Send-Q Local Address           Foreign Address         State      
tcp        0      0 0.0.0.0:8187            0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:988             0.0.0.0:*               LISTEN
```
`ss -a` use `local address` and `peer address`
```
Netid             State                  Recv-Q              Send-Q                                                                               Local Address:Port                                                Peer Address:Port           Process                                                                         
nl                UNCONN                 0                   0                                                                                             rtnl:-1048856912                                                     *
```
We already used `remote` port concept in tcplife and tcpstates, so using `local` and `remote` addresses should be okay?

**Pull request merged**